e2e-test/src/test/java/io/cdap/plugin/file/stepsdesign/FileConnector.java

```diff
@@ -0,0 +1,379 @@
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 5);
+    wait.until(ExpectedConditions.invisibilityOf(CdfStudioLocators.statusBanner));
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Run the Pipeline in Runtime")
+  public void runThePipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till pipeline is in running state")
+  public void waitTillPipelineIsInRunningState() {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 300);
+    wait.until(ExpectedConditions.or(
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Succeeded']")),
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Verify the pipeline status is {string}")
+  public void verifyThePipelineStatusIs(String status) {
+    boolean webelement = false;
+    webelement = SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']");
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("Verify Preview output schema matches the outputSchema captured in properties")
+  public void verifyPreviewOutputSchemaMatchesTheOutputSchemaCapturedInProperties() {
+    List<String> previewOutputSchema = new ArrayList<String>();
+    List<WebElement> previewOutputSchemaElements = SeleniumDriver.getDriver().findElements(
+      By.xpath("(//h2[text()='Output Records']/parent::div/div/div/div/div)[1]//div[text()!='']"));
+    for (WebElement element : previewOutputSchemaElements) {
+      previewOutputSchema.add(element.getAttribute("title"));
+    }
+    Assert.assertTrue(previewOutputSchema.equals(propertiesOutputSchema));
+  }
+
+  @Then("Close the Preview")
+  public void closeThePreview() {
+    CdfFileLocators.closeButton.click();
+    CdfStudioActions.previewSelect();
+  }
+
+  @Then("Deploy the pipeline")
+  public void deployThePipeline() {
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.pipelineDeploy, 2);
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Open Logs")
+  public void openLogs() throws FileNotFoundException, InterruptedException {
+    CdfPipelineRunAction.logsClick();
+    BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+    PrintWriter out = new PrintWriter(BeforeActions.myObj);
+    out.println(CdfPipelineRunAction.captureRawLogs());
+    out.close();
+  }
+
+  @Then("Validate successMessage is displayed")
+  public void validateSuccessMessageIsDisplayed() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Get Count of no of records transferred to BigQuery in {string}")
+  public void getCountOfNoOfRecordsTransferredToBigQueryIn(String tableName) throws IOException, InterruptedException {
+    int countRecords = GcpClient.countBqQuery(E2ETestUtils.pluginProp(tableName));
+    BeforeActions.scenario.write("**********No of Records Transferred******************:" + countRecords);
+    Assert.assertTrue(countRecords > 0);
```

Comment:
number of records transferred should be equal to the number of input records

Reply:
Changed assertion